cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/LoadingLimitsMapping.java

```diff
@@ -7,37 +7,63 @@
     private final Context context;
+    private final Map<OperationalLimitsGroup, CurrentLimitsAdder> currentLimitsAdders;
+    private final Map<OperationalLimitsGroup, ActivePowerLimitsAdder> activePowerLimitsAdders;
+    private final Map<OperationalLimitsGroup, ApparentPowerLimitsAdder> apparentPowerLimitsAdders;
 
     LoadingLimitsMapping(Context context) {
         this.context = Objects.requireNonNull(context);
+        this.currentLimitsAdders = new HashMap<>();
+        this.activePowerLimitsAdders = new HashMap<>();
+        this.apparentPowerLimitsAdders = new HashMap<>();
     }
 
-    public LoadingLimitsAdder computeIfAbsentLoadingLimitsAdder(String id, Supplier<LoadingLimitsAdder<?, ?>> supplier) {
-        return adders.computeIfAbsent(id, s -> supplier.get());
+    /**
+     * Get or create the limit adder for the given limits group and limit subclass.
+     * Different CGMES OperationalLimit can be mapped to the same IIDM LoadingLimit,
+     * hence the need to retrieve the limit adder if it has already been created before.
+     * @param limitsGroup The OperationalLimitsGroup to which the LoadingLimitsAdder shall add the limits.
+     * @param limitSubClass The operational limit subclass indicating which LoadingLimitsAdder subclass shall be used.
+     * @return The LoadingLimitsAdder for the given limits group and limit subclass.
+     */
+    public LoadingLimitsAdder getLoadingLimitsAdder(OperationalLimitsGroup limitsGroup, String limitSubClass) {
+        return switch (limitSubClass) {
+            case CgmesNames.CURRENT_LIMIT -> currentLimitsAdders.computeIfAbsent(limitsGroup, s -> limitsGroup.newCurrentLimits());
+            case CgmesNames.ACTIVE_POWER_LIMIT -> activePowerLimitsAdders.computeIfAbsent(limitsGroup, s -> limitsGroup.newActivePowerLimits());
+            case CgmesNames.APPARENT_POWER_LIMIT -> apparentPowerLimitsAdders.computeIfAbsent(limitsGroup, s -> limitsGroup.newApparentPowerLimits());
+            default -> throw new IllegalArgumentException();
+        };
     }
 
+    /**
+     * Execute the add method for all the LoadingLimitsAdder stored in this mapping class.
+     * This method shall be called after all the CGMES OperationalLimit have been converted.
+     */
     void addAll() {
-        for (Map.Entry<String, LoadingLimitsAdder<?, ?>> entry : adders.entrySet()) {
-            if (!Double.isNaN(entry.getValue().getPermanentLimit()) || entry.getValue().hasTemporaryLimits()) {
-                entry.getValue()
-                        .fixLimits(context.config().getMissingPermanentLimitPercentage(), context::fixed)
-                        .add();
+        for (Map<OperationalLimitsGroup, ? extends LoadingLimitsAdder> adder : List.of(currentLimitsAdders, activePowerLimitsAdders, apparentPowerLimitsAdders)) {
+            for (Map.Entry<OperationalLimitsGroup, ? extends LoadingLimitsAdder> entry : adder.entrySet()) {
```

Comment:
Just a minor comment we don't need the entrySet, the values are enough, so you could replace it with a for loop over `adder.values()`.

Or you could also play with streams in fact I think, starting with `Stream.of(currentLimitsAdders, activePowerLimitsAdders, apparentPowerLimitsAdders)` and then flat mapping to map values